source/auth/auth.rst

```diff
@@ -1311,47 +1309,38 @@ the form:
               type: str
               optional: true
 
-.. code:: typescript
-
-    function onOIDCRequestToken(principalName: str, serverInfo: OIDCMechanismServerStep1, timeoutSeconds: int): OIDCRequestTokenResult
+The token refresh callback must take the same arguments as the request callback, as well as the ``refreshToken`` string given by the ``IdpServerResponse``.
 
 Callbacks can be synchronous and/or asynchronous, depending on the driver
 and/or language.  Asynchronous callbacks should be preferred when other
 operations in the driver use asynchronous functions.
 
-Before calling a callback, the driver MUST acquire a lock, and verify that
-there are no valid cache credentials before calling the callback.
+Before calling a callback, the driver MUST acquire a lock, and, if there is a non-expired cached Access Token, attempt to authenticate via a JwtStepRequest before calling the callback.  The lock is released after all attempts to authenticate have failed, or a non-recoverable error was encountered.  The driver MUST ensure that credentials have not changed between when the lock was requested and when it was acquired.
```

Comment:
Probably makes sense to clarify this later unless it's trivial, but since the cache is supposed to be global, is this lock also supposed to be global? That would not be ideal imo but I'm wondering if it's implied here.

Reply:
Yeah, I also have a question about it. 
>attempt to authenticate via a JwtStepRequest before calling the callback

I guess this means that we need to do a server call under the lock which looks really wrong to me. Is it really meant here?

Reply:
Added "a lock unique to the cache key"